core/src/main/java/tech/tablesaw/api/Table.java

```diff
@@ -1135,6 +1135,70 @@ public boolean hasNext() {
     return Streams.stream(rollingIterator(n));
   }
 
+  /**
+   * Transposes data in the table, switching rows for columns. For example, a table like this.<br>
+   * label | value1 | value2 |<br>
+   * -------------------------------<br>
+   * row1 | 1 | 2 |<br>
+   * row2 | 1.1 | 2.1 |<br>
+   * row3 | 1.2 | 2.2 |<br>
+   *
+   * <p>Is transposed into the following<br>
+   * label | row1 | row2 | row3 |<br>
+   * -------------------------------------<br>
+   * value1 | 1 | 1.1 | 1.2 |<br>
+   * value2 | 2 | 2.1 | 2.2 |<br>
+   *
+   * @return The transposed table
+   */
+  public Table transpose() {
+    Table transposed = Table.create(this.name);
+    Column firstColumn = this.column(0);
```

Comment:
This probably won't work on an empty table. It'd be nice to handle that case a bit better by just returning `transposed` first if there are no columns

Also, this variable isn't used until awhile later. It'd be nicer to declare it right before it's used. Or actually I'd probably just inline it:

```
for (String label : this.column(0).asStringColumn()) {
```

Reply:
Good catch. I've added a test for the empty table and made the change